PowerEditor/src/WinControls/Preference/preferenceDlg.cpp

```diff
@@ -5596,13 +5611,44 @@ intptr_t CALLBACK SearchingSubDlg::run_dlgProc(UINT message, WPARAM wParam, LPAR
 
 		case WM_COMMAND:
 		{
+			if ((LOWORD(wParam) == IDC_INSELECTION_THRESHOLD_EDIT) &&
+				(HIWORD(wParam) == EN_CHANGE))
+			{
+				constexpr int stringSize = 5;
+				wchar_t str[stringSize]{};
+				::GetDlgItemText(_hSelf, IDC_INSELECTION_THRESHOLD_EDIT, str, stringSize);
+
+				if (lstrcmp(str, L"") == 0)
+				{
+					::SetDlgItemInt(_hSelf, IDC_INSELECTION_THRESHOLD_EDIT, nppGUI._inSelectionAutocheckThreshold, FALSE);
+					return FALSE;
+				}
+
+				UINT newValue = ::GetDlgItemInt(_hSelf, IDC_INSELECTION_THRESHOLD_EDIT, nullptr, FALSE);
+
+				if (static_cast<int>(newValue) == nppGUI._inSelectionAutocheckThreshold)
+				{
+					return FALSE;
+				}
+
+				if (newValue > FINDREPLACE_INSELECTION_THRESHOLD_DEFAULT)
+				{
+					::SetDlgItemInt(_hSelf, IDC_INSELECTION_THRESHOLD_EDIT, FINDREPLACE_INSELECTION_THRESHOLD_DEFAULT, FALSE);
+					newValue = FINDREPLACE_INSELECTION_THRESHOLD_DEFAULT;
+				}
+
+				nppGUI._inSelectionAutocheckThreshold = newValue;
+
+				return TRUE;
+			}
+
 			switch (wParam)
 			{
 				case IDC_CHECK_FILL_FIND_FIELD_WITH_SELECTED:
 				{
 					nppGUI._fillFindFieldWithSelected = isCheckedOrNot(IDC_CHECK_FILL_FIND_FIELD_WITH_SELECTED);
-					::EnableWindow(::GetDlgItem(_hSelf, IDC_CHECK_FILL_FIND_FIELD_SELECT_CARET), nppGUI._fillFindFieldWithSelected ? TRUE :FALSE);
-					if (!nppGUI._fillFindFieldWithSelected) 
+					::EnableWindow(::GetDlgItem(_hSelf, IDC_CHECK_FILL_FIND_FIELD_SELECT_CARET), nppGUI._fillFindFieldWithSelected ? TRUE : FALSE);
+					if (!nppGUI._fillFindFieldWithSelected)

```

Comment:
Hmm, I didn't mean to affect whitespace here...I hope it is OK and doesn't contribute too much to meaningless changes to review...